src/main/java/com/rcv/ResultsWriter.java

```diff
@@ -428,32 +436,46 @@ private void generateSummaryJson(String outputPath,
 
     // root outputJson dict will have two entries:
     // results - vote totals, transfers, and candidates elected / eliminated
-    // TODO: add needed config info
     // config - global config into
     HashMap<String, Object> outputJson = new HashMap<>();
+    // config will contain contest configuration info
+    HashMap<String, Object> configData = new HashMap<>();
+    // add config header info
+    configData.put("Contest", config.getContestName());
+    configData.put("Jurisdiction", config.getContestJurisdiction());
+    configData.put("Office", config.getContestOffice());
+    configData.put("Date", config.getContestDate());
+    configData.put("Threshold", winningThreshold.toString());
+    if (precinct != null && !precinct.isEmpty()) {
+      configData.put("Precinct", precinct);
+    }
     // results will be a list of round data objects
     ArrayList<Object> results = new ArrayList<>();
     // for each round create objects for json serialization
     for(int round = 1; round <= numRounds; round++) {
       // container for all json data this round:
       HashMap<String, Object> roundData = new HashMap<>();
       // add round number (this is implied by the ordering but for debugging we are explicit)
-      roundData.put("round",round);
+      roundData.put("Round", round);
+      // add actions if this is not a precinct summary
+      if (precinct == null || precinct.isEmpty()) {
       // actions is a list of one or more action objects
-      ArrayList<Object> actions = new ArrayList<>();
-      // add any elected actions
-      addActionObjects("elected", roundToWinningCandidates.get(round), round, actions);
-      // add any elimination actions
-      addActionObjects("eliminated", roundToEliminatedCandidates.get(round), round, actions);
-      // add action objects
-      roundData.put("actions", actions);
+        ArrayList<Object> actions = new ArrayList<>();
```

Comment:
nit: LinkedList is a little cheaper, I think

Reply:
you think?

Reply:
Well, an ArrayList is a dynamically-sized array, whereas a LinkedList is just a linked list. If you need to be able to access elements by index in constant time (vs. linear), ArrayList is definitely better. But I've always assumed that if you're just building a list that you're going to iterate over later, LinkedList is preferable because it's a simpler data structure. I could be wrong. In any case, it hardly matters at a tiny scale like this.

If you want to read way more on the subject: https://stackoverflow.com/questions/10656471/performance-differences-between-arraylist-and-linkedlist